app/entry.server.tsx

```diff
@@ -136,3 +147,27 @@ function handleBrowserRequest(
     setTimeout(abort, streamTimeout + 1000);
   });
 }
+
+/**
+ * Flushes pending Sentry events with a 2 second timeout and in a way that cannot create unhandled promise rejections.
+ */
+async function flushSafelyWithTimeout(): Promise<void> {
+  try {
+    await Sentry.flush(2000);
+  } catch (error) {
+    console.error('Error while flushing events to Sentry', {
+      cause: error,
+    });
+  }
+}
+
+export const handleError: HandleErrorFunction = async (error, { request }) => {
+  // React Router may abort some interrupted requests, don't log those
+  if (!request.signal.aborted) {
+    Sentry.captureException(error);
+
+    console.error(error);
+
+    vercelWaitUntil(flushSafelyWithTimeout());
```

Comment:
this was the key part for server errors. vercel was stopping the execution before the error was sent to sentry